test/smime-certs/mksmime-certs.sh

```diff
@@ -69,8 +69,7 @@ CN="Test S/MIME EE EC #3" $OPENSSL req -config ca.cnf -noenc \
 $OPENSSL x509 -req -in req.pem -CA smroot.pem -days 3600 \
 	-extfile ca.cnf -extensions usr_cert -CAcreateserial >>smec3.pem
 # Create X9.42 DH parameters.
-$OPENSSL genpkey -genparam -algorithm DH -pkeyopt dh_paramgen_type:2 \
-	-out dhp.pem
+$OPENSSL genpkey -genparam -algorithm DHX -out dhp.pem
```

Comment:
Uhm.. Should we have 2 separate tests instead rather than replacing the existing one to make it work?

Reply:
I am not sure what that would bring. Do you mean keeping the old parameters as negative test case for the fips? That would require further changes in the test-cms recipe which I'd prefer not to do in this PR.

Reply:
Well, the DHX algorithm choice for the cli is underdocumented, and not what a user would expect to use to "Create X9.42 DH parameters", so I believe we should test a real use case rather than alter the existing use case to hammer it into working within FIPS unexpected requirements.

If this means we need a separate PR to tackle the problems that the previous version of this command in the test highlights, we should open that, address the issue and make this dependent on its resolution. 

I admit I have limited knowledge in the CMS and SMIME area, my warning bells just started tingling the moment we need to replace a `genpkey - algorithm DH` with `DHX` which to this point I thought was just an internal conceptual distinction.

Reply:
This is what #13871 is about. Please note that this change is only for the parameter generation for the test. Not the test itself.

Reply:
My point is that we should _replace_ `genpkey -genparam -algorithm DH -pkeyopt dh_paramgen_type:2` with `genpkey -genparam -algorithm DHX` only if that was a mistake and the first kind of parameters was never supposed to work in the test.

We should _add_ `genpkey -genparam -algorithm DHX` if _also_ this kind of parameters is supposed to work in this test.

What we shouldn't be doing is reduce the test coverage by removing the `DH` option because we need extra work to distinguish between test cases that should succeed or fail in `default` or `fips` settings.

> Please note that this change is only for the parameter generation for the test. Not the test itself.

Well, by providing 2 different "sets" of parameters, even if they are generated once instead of at every test run, we are changing what codepaths the test is exercising, aren't we?

Reply:
No, the mksmime-certs.sh is not part of the test. The generated parameters are of the same type as the old ones, they are just big enough to pass the FIPS checks. That I used genpkey -genparam -algorithm DHX as a convenient way to generate such parameters instead of trying to find other way with -algorithm DH (if it is at all possible given #13871) is irrelevant for this PR. As the issues with the parameter generation changes are supposed to be covered by the #13871 I do not see reason to further complicate this PR (which still needs work on the decryption side).